### DISCUSSION TITLE
Issue 'ValueError: operands could not be broadcast together with shapes (2,3,1) (3,1,3) '

### DISCUSSION THREAD

**Discussion by joshuapeters0210** [pgmpy/pgmpy#1500] [Q&A]
Basically I wish to set some evidence in my BN ie (if this node was this, and this node was this what would be the updated cpds).

I can't find a decent way to do this, whenever I try to use

infer_non_adjust = VariableElimination(macid)

print(infer_non_adjust.query(variables=["UD"], evidence={"UI": 'No'}))

I get the following error.

'ValueError: operands could not be broadcast together with shapes (2,3,1) (3,1,3) '

However on a test project I did it worked fine, as far as I can tell there isnt too much difference between the two codes.

My questions are, what could be causing this issue and is there an alternate way to find this information out.

As a point this is my first time using python, im not particulary good with this and this is simply for data analysis of other work, so sorry if I struggle with the basics (i'm trying to learn)

**Comment by ankurankan** [comment]
@joshuapeters0210 It's hard to say without looking at the rest of your code but the first thing I would check is that the variables used for query and evidence exist in the learned model and also that the state names exist in the CPD.
An alternate way would be to use approximate inference based on sampling: https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/inference/ApproxInference.py#L71 . The query method here is quite similar to the VariableElimination one but uses a different approach to compute the value.

**Comment by joshuapeters0210** [comment]


**Reply by joshuapeters0210** [reply]
@ankurankan I've tried your suggested fixes but unfortunately neither work. I will try use the alternative method and let you know. Thanks!
